src/httpation_uri.erl

```diff
@@ -203,11 +191,16 @@ consume_port(_, _, _) ->
     {error, port}.
 
 
+consume_path("", _Parts, "", URI) ->
+    {ok, URI#uri{path = ["/"]}};
 consume_path(Acc, Parts, "", URI) ->
     Part = lists:reverse(Acc),
     NewParts = [Part | Parts],
     Path = lists:reverse(NewParts),
     {ok, URI#uri{path = Path}};
+consume_path(_Acc, Parts, [$/ | [$/ | Rest]], URI) ->
```

Comment:
Two things here:

1. What is the difference between encountering `[$/ | [$/ | Rest]]` and encountering `[$/, $/ | Rest]`? Which is more obvious and easier to read? *Is* there a technical difference?
2. Why would we *not* need to know the value of `Acc` in this case? Why are we labeling it with an underscore if we don't need it *and* its value and name are obvious from the surrounding code? (I'm not particularly religious about `_SomeValue` vs `_` -- but if we are going to use an "I don't care" value then it means *you really don't care*. If your code is confusing without unused labels then your code is probably too confusing for a maintainer even if the labels are there.)

More to the point -- in what cases will we encounter `(Acc, Parts, [$/, $/ | Rest], URI)` that are *not* already covered correctly by the case `("", Parts, [$/ | Rest], URI)`?

This parsing algorithm is canonically defined as a parser that *does not need* lookahead. Adding lookahead every time we encounter a slash adds overhead that will never match if the function is designed properly. What happens, for example, if we encounter triple or quadruple slashes?